hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -346,14 +353,18 @@ public Fees calculateFees(@NonNull final FeeContext feeContext) {
      * @param keySize the size of the key
      * @return the calculated base size
      */
-    private static long baseSizeOf(final CryptoUpdateTransactionBody op, final long keySize) {
-        return BASIC_ENTITY_ID_SIZE
+    private static long baseSizeOf(
+            final CryptoUpdateTransactionBody op, final long keySize, final boolean unlimitedAutoAssociationsEnabled) {
+        var baseSize = BASIC_ENTITY_ID_SIZE
                 + op.memoOrElse("").getBytes(StandardCharsets.UTF_8).length
                 + (op.hasExpirationTime() ? LONG_SIZE : 0L)
                 + (op.hasAutoRenewPeriod() ? LONG_SIZE : 0L)
                 + (op.hasProxyAccountID() ? BASIC_ENTITY_ID_SIZE : 0L)
-                + (op.hasMaxAutomaticTokenAssociations() ? INT_SIZE : 0L)
                 + keySize;
+        if (!unlimitedAutoAssociationsEnabled) {
+            baseSize += op.hasMaxAutomaticTokenAssociations() ? INT_SIZE : 0L;
+        }
```

Comment:
This doesn't need to check `!unlimitedAutoAssociationsEnabled` right ? As long as something is set here we should charge fees.